node/pkg/wfetcher/common/utils.go

```diff
@@ -0,0 +1,88 @@
+package common
+
+import (
+	"context"
+	"encoding/json"
+	"math"
+	"strconv"
+	"strings"
+
+	"bisonai.com/orakl/node/pkg/db"
+	"github.com/rs/zerolog/log"
+)
+
+/*
+Generates two types of maps with different keys for the same feed ID.
+The combined map has keys like "BTCUSD", and the separated map has keys like "BTC-USD".
+The reason for this is that different exchanges use different naming conventions.
+*/
+func GetWssFeedMap(feeds []Feed) map[string]FeedMaps {
+	feedMaps := make(map[string]FeedMaps)
+	for _, feed := range feeds {
+		if !strings.Contains(feed.Name, "wss") {
```

Comment:
I think we could do it less brute force. `Feed` could have `sourceType` or some other field to specify where it is coming from. Combining exchange, source type and symbol name in a single string (e.g. `binance-wss-BTC-USDT`) is very error prone and does not show how the solution is expected to be extended.

Reply:
We have to manually write inside feed definitions anyways, first approach was to write provider name inside the definition but I realized it was duplicate of information and ended up here. I'm not sure how adding info in definition is less brute force? it should be parsed inside again through json.unmarshal and valued should be checked

Reply:
In the long term it does not matter when we move everything to ws. If we move websocket definitions to separate array within adapter config, then we don't have to specify the source type anywhere for example.

```json
{
  "adapterHash": "0xf15e595e641f1447b5ad1c128007574b329e00937e6db3857aff1487345ed5f6",
  "name": "ADA-USDT",
  "decimals": 8,
  "feeds": [
      ....
  ],
  "wsFeeds": [
      ....
  ],
}

Reply:
your suggestion seems to be another big update, I'll handle it from further PRs. [issue](https://github.com/Bisonai/orakl/issues/1538) has been created